code/game/machinery/teleporter/console.dm

```diff
@@ -0,0 +1,189 @@
+
+
+/obj/machinery/computer/teleporter/proc/clear_pad()
+	if (!pad)
+		return
+	GLOB.destroyed_event.unregister(pad, src, /obj/machinery/computer/teleporter/proc/lost_pad)
+	pad = null
+	set_active(FALSE)
+
+
+/obj/machinery/computer/teleporter/proc/lost_pad()
+	audible_message(SPAN_WARNING("\The [src] buzzes, \"Pad missing.\""))
+	clear_pad()
+
+
+/obj/machinery/computer/teleporter/proc/set_pad(obj/machinery/tele_pad/_pad)
+	if (pad == _pad)
+		return
+	clear_pad()
+	pad = _pad
+	GLOB.destroyed_event.register(pad, src, /obj/machinery/computer/teleporter/proc/lost_pad)
+
+
+/obj/machinery/computer/teleporter/proc/clear_target()
+	if (!target)
+		return
+	GLOB.destroyed_event.unregister(target, src, /obj/machinery/computer/teleporter/proc/lost_target)
+	target = null
+	set_active(FALSE)
+
+
+/obj/machinery/computer/teleporter/proc/lost_target()
+	audible_message(SPAN_WARNING("\The [src] buzzes, \"Target lost.\""))
+	clear_target()
+
+
+/obj/machinery/computer/teleporter/proc/set_target(atom/_target)
+	if (target == _target)
+		return
+	clear_target()
+	target = _target
+	GLOB.destroyed_event.register(target, src, /obj/machinery/computer/teleporter/proc/lost_target)
+
+
+/obj/machinery/computer/teleporter/proc/set_active(_active, notify)
+	var/effective = _active && target && projector && pad
+	if (active == effective)
+		return
+	active = effective
+	if (notify && effective)
+		if (active)
+			visible_message(SPAN_NOTICE("The teleporter sparks and hums to life."))
+		else
+			visible_message(SPAN_WARNING("The teleporter sputters and fails."))
+	if (projector)
+		projector.update_icon()
+	if (pad)
+		pad.update_icon()
+
+
+/obj/machinery/computer/teleporter/proc/get_targets()
+	var/list/ids = list()
+	var/list/result = list()
+	for (var/obj/item/radio/beacon/B in world)
+		if (QDELETED(B) || !B.functioning || !isPlayerLevel(B.z))
+			continue
+		var/area/A = get_area(B)
+		if (!A)
+			continue
+		result["[A.name] \[[++ids[A]]\]"] = B
+	for (var/obj/item/implant/tracking/T)
+		if (QDELETED(T) || !T.implanted || !ismob(T.loc))
+			continue
+		var/mob/M = T.loc
+		if (M.stat == DEAD && world.time > M.timeofdeath + 15 MINUTES)
+			continue
+		if (!isPlayerLevel(M.z))
+			continue
+		result["[M.name] \[[++ids[M]]\]"] = T
+	return result
+
+
+/obj/machinery/computer/teleporter/power_change()
+	. = ..()
+	if (!.)
+		return
+	if (stat & NOPOWER)
+		clear_target()
+
+
+/obj/machinery/computer/teleporter/interface_interact(mob/user)
+	if (!projector || !pad)
+		var/data_search = alert(user, "Projector or Pad missing. Search?", "Teleporter", "Yes", "No")
+		if (isnull(data_search) || !CanInteract(user, DefaultTopicState()))
+			return TRUE
+		if (data_search == "Yes")
+			update_refs()
+		return TRUE
+	var/message = "Teleporter [!target ? "Idle" : !active ? "Locked" : "Engaged"]\n[!target ? "" : "\[[get_area(target)]\]"]"
+	var/btn_active = active ? "Shut Down" : target ? "Start Up" : "-"
```

Comment:
Please either break this into if/else blocks, or add parens to the subclauses, because this line and the line above are painful to read.